course/flow.py

```diff
@@ -1612,7 +1612,9 @@ def add_buttons_to_form(form, fpctx, flow_session, permissions):
     return form
 
 
-def create_flow_page_visit(request, flow_session, page_data):
+@retry_transaction_decorator()
+def create_flow_page_visit(request, flow_session, page_data,
+        answer_data, is_submitted_answer):
```

Comment:
I'm not in principle against centralizing `FlowPageVisit` creation, but the way it's written forces even the logging of (fairly expendable) `GET` visits into a (hugely expensive, retry-hardened) `SERIALIZABLE` database transaction. The `GET` path should stay non-transactional. Maybe move split the function out into `create_flow_page_visit_transactional` and `create_flow_page_visit_base`?